app/src/lib/stores/app-store.ts

```diff
@@ -1968,7 +1971,42 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return
     }
 
+    const account = getAccountForRepository(this.accounts, repository)
```

Comment:
:art: Something to keep in mind as you're working through this - I'd love for this logic to be reusable so we can eventually use this in `BackgroundFetcher`